addons/udes_stock/models/stock_move.py

```diff
@@ -1,10 +1,31 @@
 # -*- coding: utf-8 -*-
-from odoo import models, api, _
+from odoo import models, fields, api, _
+from odoo.addons import decimal_precision as dp
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    u_initial_demand = fields.Float("Initial Demand", digits="Product Unit of Measure")
+
+    product_uom_qty = fields.Float(string="Quantity")
+
+    @api.model
+    def create(self, vals):
+        if "u_initial_demand" not in vals:
+            vals["u_initial_demand"] = vals.get("product_uom_qty")
+        return super().create(vals)
+
+    def _do_unreserve(self):
+        """
+        Override _do_unreserve to remove extra quantities added to initial_demand when
+        unreserving a package with full package reservation.
+        """
+        res = super()._do_unreserve()
+        for move in self:
+            move.write({"product_uom_qty": move.u_initial_demand})
```

Comment:
This is where we should delete moves where its initial demand is 0.

If this is about full package reservation, why it is not in the other module?

Reply:
The idea is that in `udes_stock` we just add a field that will hold whatever the quantity was first set to. Atm only full package reservation will change the quantity from the initial value, but we might have other functionality that does so in future.

We don't need to delete any moves where the initial demand is 0, Odoo's merging of moves on confirm isn't affected by this. This is just for info only.

Reply:
So imagine you have a picking for 5 apples, but only pack available has 20. After reserving stock the quantity is set from 5 to 20. But when unreserving we now set the quantity back to the initial demand.

Reply:
I just spotted the doc-string though, @caleb-shelton please edit this so it doesn't mention full package reservation. :)